src/client/Microsoft.Identity.Client/ManagedIdentity/IProbe.cs

```diff
@@ -0,0 +1,13 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Identity.Client.ManagedIdentity
+{
+    internal interface IProbe
```

Comment:
The only reason to introduce an interface is if you're going to mock it, but it doesn't seem to be the case here. You don't need to mock this object, as you can rely on HTTP mocking.